src/sas/sascalc/file_converter/bsl_loader.py

```diff
@@ -76,8 +76,67 @@ def __init__(self, filename):
         if not is_valid:
```

Comment:
For now, I have added the header_file.close() before raising the error, however I did not write any of the code in __init__ and cannot run or test bsl_loader properly so am hesitant about making further changes, but can if necessary.

Reply:
Understood.  Some options:

(1) take the code from the 4.x source that runs the converter and turn it into a test case

(2) backport the bls changes to the 4.x code so you can run the converter from the GUI

(3) port the converter dialog from 4.x to 5.x

(4) change the converter into a data loader so that you don't need a converter dialog.

Reply:
Alright, I'll look at backporting it to 4.x and look at improving the exception handling in init.